packages/cmf/src/cmfConnect.js

```diff
@@ -142,7 +142,9 @@ export default function cmfConnect({
 						this.context
 					);
 				}
-				if (!keepComponentState) {
+				// if the props.keepComponentState is present we have to stick to it
+				if (this.props.keepComponentState === false ||
+					(this.props.keepComponentState === undefined && !keepComponentState)) {
 					this.props.deleteState();
```

Comment:
well i did that cause i wanted to manage especially the false value, because if the props.keepComponentState is undefined  it still has to go check for the default flag (the one on container)